test/python/transpiler/test_preset_passmanagers.py

```diff
@@ -167,6 +167,40 @@ def test_symmetric_coupling_map(self, level):
         self.assertIn('ApplyLayout', self.passes)
         self.assertNotIn('CheckCXDirection', self.passes)
 
+    @data(0, 1, 2, 3)
+    def test_inital_layout_fully_connected_cm(self, level):
+        """Honor initial_layout when coupling_map=None
+        See: https://github.com/Qiskit/qiskit-terra/issues/5345
+        """
+        qr = QuantumRegister(2, 'q')
+        qc = QuantumCircuit(qr)
+        qc.h(qr[0])
+        qc.cx(qr[0], qr[1])
+
+        transpiled = transpile(qc, initial_layout=[0, 1], optimization_level=level,
+                               callback=self.callback)
+
+        self.assertIn('SetLayout', self.passes)
+        self.assertIn('ApplyLayout', self.passes)
+        self.assertEqual(len(transpiled._layout), 2)
+
+    @data(0, 1, 2, 3)
+    def test_partial_layout_fully_connected_cm(self, level):
+        """Honor initial_layout (partially defined) when coupling_map=None
+        See: https://github.com/Qiskit/qiskit-terra/issues/5345
+        """
+        qr = QuantumRegister(2, 'q')
+        qc = QuantumCircuit(qr)
+        qc.h(qr[0])
+        qc.cx(qr[0], qr[1])
+
+        transpiled = transpile(qc, initial_layout=[4, 2], optimization_level=level,
+                               callback=self.callback)
```

Comment:
so what does `transpiled` look like here? is it a 2 qubit circuit or a 5 qubit circuit?

Reply:
5, see line 202

Reply:
I think taking the max of the initial layout makes sense for these situations.